internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/JobStatusQueryResponse.java

```diff
@@ -135,15 +92,17 @@ public void setExtraDetails(Map<String, String> extraDetails) {
     this.extraDetails = extraDetails;
   }
 
-  public static JobStatusQueryResponse createErrorResponse(String errorMessage) {
-    JobStatusQueryResponse response = new JobStatusQueryResponse();
-    response.setError(errorMessage);
-    return response;
+  public void setUncompletedPartitions(List<UncompletedPartition> uncompletedPartitions) {
+    this.uncompletedPartitions = uncompletedPartitions;
+  }
+
+  public List<UncompletedPartition> getUncompletedPartitions() {
+    return uncompletedPartitions;
   }
 
   public String toString() {
     return JobStatusQueryResponse.class.getSimpleName() + "(\n" + "version: " + version + ",\n" + "status: " + status
         + ",\n" + "statusDetails: " + statusDetails + ",\n" + "extraInfo: " + extraInfo + ",\n" + "extraDetails: "
-        + extraDetails + ",\n" + super.toString() + ")";
+        + extraDetails + ",\n" + "uncompletedPartitions: " + uncompletedPartitions + ",\n" + super.toString() + ")";
```

Comment:
Just wondering if it will be problematic excessive logging  for those stores who have 1k+ partition?

Reply:
This sounds a valid concern. I remove uncompletedPartitions from JobStatusQueryResponse#toString(). Data recovery monitoring does not need the list to be present in the toString() method.